### PR TITLE
feat(iot): Add ability to set user device certificate

### PR DESCRIPTION
**Pull request by grom-42**

### CODE REVIEW

File: scaleway/resource_iot_device.go [hunk 1]
```diff
@@ -141,18 +130,31 @@ func resourceScalewayIotDevice() *schema.Resource {
 					Schema: map[string]*schema.Schema{
 						"crt": {
 							Type:        schema.TypeString,
+							Optional:    true,
 							Computed:    true,
```
Comment (Monitob) [inline scaleway/resource_iot_device.go:134]:
```suggestion
							Optional:    true,
							ForceNew:      true,
							Computed:    true,
```
 Please consider add ForceNew

File: scaleway/resource_iot_device.go [hunk 2]
```diff
@@ -141,18 +130,31 @@ func resourceScalewayIotDevice() *schema.Resource {
 					Schema: map[string]*schema.Schema{
 						"crt": {
 							Type:        schema.TypeString,
```
Comment (Monitob) [inline scaleway/resource_iot_device.go:132]:
```suggestion
							Type:        schema.TypeString,
							Required:    true,
```
This should be required

File: scaleway/resource_iot_device.go [hunk 3]
```diff
@@ -319,11 +335,32 @@ func resourceScalewayIotDeviceRead(ctx context.Context, d *schema.ResourceData,
 		_ = d.Set("message_filters", []map[string]interface{}{mf})
 	}
 
+	////
+	// Read Device certificate
+	////
+
+	// As we cannot read the key, we get back it from cache and do not change it.
```
Comment (Monitob) [inline scaleway/resource_iot_device.go:342]: @grom-42 I can't find any code related with your comment. Can you give an inside?

File: scaleway/resource_iot_device.go [hunk 4]
```diff
@@ -242,12 +244,26 @@ func resourceScalewayIotDeviceCreate(ctx context.Context, d *schema.ResourceData
...
-		"crt": res.Certificate.Crt,
-		"key": res.Certificate.Key,
+	// If user certifcate is provided.
+	if devCrt, ok := d.GetOk("certificate.0.crt"); ok {
+		// Set user certificate to device.
+		// It cannot currently be added in the create device request.
+		_, err := iotAPI.SetDeviceCertificate(&iot.SetDeviceCertificateRequest{
+			Region:         region,
+			DeviceID:       res.Device.ID,
+			CertificatePem: devCrt.(string),
```
Comment (Monitob) [inline scaleway/resource_iot_device.go:254]: Please add a safer method to cast this. e.g.
`s, ok := devCrt.(string)
if !ok {
    //  manage your error
}`

File: scaleway/resource_iot_device.go [hunk 5]
```diff
@@ -319,11 +335,32 @@ func resourceScalewayIotDeviceRead(ctx context.Context, d *schema.ResourceData,
...
+			Region:   region,
+			DeviceID: deviceID,
+		}, scw.WithContext(ctx))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		// Set device certificate.
+		cert := map[string]interface{}{
+			"crt": devCrt.CertificatePem,
+			"key": devCrtKey.(string),
```
Comment (Monitob) [inline scaleway/resource_iot_device.go:354]: idem

File: scaleway/resource_iot_device.go (hunk 1 above)
Comment (grom-42) [reply scaleway/resource_iot_device.go:134]: As discussed we do not want to trigger the certificate re-creation if the user updates its certificate.

File: scaleway/resource_iot_device.go (hunk 2 above)
Comment (grom-42) [reply scaleway/resource_iot_device.go:132]: As discussed, do not provide a custom certificate is a legit use case, so it is not mandatory.

File: scaleway/resource_iot_device.go (hunk 4 above)
Comment (grom-42) [reply scaleway/resource_iot_device.go:255]: As discussed, for coherency with the basecode we will let as is for now and take it in consideration for future improvements.

File: scaleway/resource_iot_device.go (hunk 5 above)
Comment (grom-42) [reply scaleway/resource_iot_device.go:355]: idem :)

**Review by Monitob — APPROVED** [review]

**Pull request merged**